app/src/main/java/org/schabi/newpipe/util/external_communication/ShareUtils.java

```diff
@@ -203,36 +127,82 @@ private static void openAppChooser(@NonNull final Context context,
                 chooserIntent.addFlags(permFlags);
             }
         }
-        context.startActivity(chooserIntent);
+
+        try {
+            context.startActivity(chooserIntent);
+            return true;
+        } catch (final ActivityNotFoundException e) {
+            // The system chooser should be always available, so this shouldn't happen
+            Toast.makeText(context, R.string.no_app_to_open_intent, Toast.LENGTH_SHORT)
+                    .show();
+
+            return false;
+        }
     }
 
     /**
-     * Get the default app package name.
+     * Open the given URL in its default app or in an app chooser.
+     *
+     * @param context    the context to use
+     * @param url        the URL to browse
+     * @param useChooser whether to use a system chooser instead of opening the link in the default
+     *                   app or browser (the default app associated with this link or the default
+     *                   browser if there is no default app will be only shown in the system
+     *                   chooser on Android 12 and higher)
+     * @return whether the URL can be opened
+     */
+    public static boolean openUrlInBrowser(@NonNull final Context context,
+                                           final String url,
+                                           final boolean useChooser) {
```

Comment:
Also here. I think it would be better to create two different methods, so that it is clear what's happening just by reading the code (no need to look up parameter documentation).
- `openUrlInDefaultBrowser`
- `openUrlInBrowserWithChooser`

Reply:
Not a bad idea, but I would use `openUrlInDefaultApp` and `openUrlWithAppChooser`, as we do not look at a browser anymore.

Reply:
Right